activiti-cloud-services-org/activiti-cloud-services-org-service/src/main/java/org/activiti/cloud/services/organization/validation/ProcessModelValidator.java

```diff
@@ -94,11 +112,63 @@ public void validateModelContent(byte[] bytes) {
+                                                                     ValidationContext validationContext) {
+        List<String> availableImplementations = getAvailableImplementations(validationContext);
+
+        return getTasksStream(bpmnModel,
+                              ServiceTask.class)
+                .map(serviceTask -> validateServiceTaskImplementation(serviceTask,
+                                                                      availableImplementations))
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .collect(Collectors.toList());
+    }
+
+    private <T extends Task> Stream<T> getTasksStream(BpmnModel bpmnModel,
+                                                      Class<T> taskType) {
+        return bpmnModel.getProcesses()
+                .stream()
                 .flatMap(process -> process.getFlowElements().stream())
-                .filter(flowElements -> flowElements.getClass().isAssignableFrom(UserTask.class))
-                .map(UserTask.class::cast)
+                .filter(element -> taskType.isAssignableFrom(element.getClass()))
+                .map(taskType::cast);
+    }
+
+    private List<String> getAvailableImplementations(ValidationContext validationContext) {
+        return validationContext.getAvailableModels()
+                .stream()
+                .filter(model -> ConnectorModelType.NAME.equals(model.getType()))
+                .map(this::concatImplementationAndAction)
+                .flatMap(Collection::stream)
+                .collect(Collectors.toList());
+    }
+
+    private List<String> concatImplementationAndAction(Model model) {
+        return connectorModelContentConverter.convertToModelContent(model.getContent().getBytes())
+                .filter(Objects::nonNull)
+                .get()
+                .getActions()
+                .values()
+                .stream()
+                .map(connectorModelAction -> StringUtils.isEmpty(connectorModelAction.getName()) ? model.getName() : model.getName() + "." + connectorModelAction.getName())
+                .collect(Collectors.toList());
+    }
+
+    private Optional<ValidationError> validateServiceTaskImplementation(ServiceTask serviceTask,
+                                                                        List<String> availableImplementations) {
+        ValidationError validationError = new ValidationError();
+        validationError.setProblem(INVALID_SERVICE_IMPLEMENTATION_PROBLEM);
+        validationError.setValidatorSetName(SERVICE_USER_TASK_VALIDATOR_NAME);
+        validationError.setDefaultDescription(String.format(INVALID_SERVICE_IMPLEMENTATION_DESCRIPTION,
+                                                            serviceTask.getId()));
+
+        return availableImplementations.stream().anyMatch(implementation -> serviceTask.getImplementation().equals(implementation)) ?
```

Comment:
If we are not in a project validation context, availableImplementations will be always empty and the validation will always fail. I think this case should be treated differently.